### ISSUE TITLE
Stripping used closure parameters (version 0.23)

### ISSUE THREAD

**Issue by mergesort** [nicklockwood/SwiftFormat#80]
I'm running into an issue, where the return type is being stripped out incorrectly from a closure.

```
Something.login().then { (userID, password) -> Promise<String> in
    // Stuff here
    // Stuff here
    return Something.getAccessToken(userID: userID, password: password) // Returns Promise<String>
}.then { accessToken -> Void in
    // Stuff here
}.catch { error in
    print(error)
}
```

The issue is that the `Promise<String>` is being converted to `Promise<_>` for some reason, even though it's being used. This compiled before (I upgraded from 0.19) and now keeps inserting that incorrectly.

Any help would be appreciated, happy to provide more info as necessary.

**Comment by kdubb** [comment]
If it adds any insight this happens for us as well, specifically on `Promise<>`

**Comment by nicklockwood** [comment]
Sorry for not responding sooner, not sure how I missed this

 I'll try to get a fix up asap. In the meantime, you can work around it by adding `--disable unusedArguments` to the swiftformat command.

**Comment by nicklockwood** [comment]
Fixed in 0.23.1
